src/builtins/core/zoneddatetime.rs

```diff
@@ -251,6 +252,7 @@ pub struct ZonedDateTime {
     instant: Instant,
     calendar: Calendar,
     tz: TimeZone,
+    cached_offset: UtcOffset,
```

Comment:
thought: my temptation looking at this is to have the time zone and cached `UtcOffset` associated with one another in one type.

Not entirely sure it's the best idea, but I think the two should be tied together.